Stride-Labs/stride
# Docker net multiple host

Closes: #XXX

## What is the purpose of the change

* WIth the addition of new host zones, the localnet setup has become increasingly more complex, warranting a move back to docker. This PR generalizes the setup for host zones such that adding a new host zone should be a simple process that can be done in <15 minutes.
* Startup time should be ~1.5 min faster than localnet 
* Shoutout to Vishal for all the work he did on adding host zones to localnet - it made it really easy to port this over to docker (and saved me a lot of headaches) 
* **NOTE**: This PR is branched off a different dockernet [PR](https://github.com/Stride-Labs/stride/pull/196), please review that first

## Brief Changelog
- Generalized the build (`scripts/build.sh`), chain init (`init_chain.sh`), startup (`start_network.sh`), relayer (`init_relayers.sh, create_channels.sh`), and register host zone (`register_host.sh`) scripts for dockernet
- Pointed Stride and Gaia to these new scripts
- Added Juno and Osmo to dockernet
- Triggered commands in parallel to bring down the startup time
- Added instructions for adding a new host zone in `scripts/README.md`

## Testing and Verifying
* Tested with firehose on each host zone
```
2022/08/22 10:18:43 TOTAL: 100/100(100.00) Success
```
* To start, run 
```
make start-docker build=sgjo
```
## Testing with firehose:
 * Make sure the denom's in `flows/bank.go` line up with `scripts/vars.sh` (from the Stride repo)
```
IbcAtom = "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
IbcOsmo = "ibc/13B2C536BB057AC79D5616B8EA1B9540EC1F2170718CAFF6F0083C966FFFED0B"
IbcJuno = "ibc/EFF323CC632EC4F747C61BCE238A758EFDB7699C3226565F7C20DA06509D59A5"
```
* Make sure the ports in `cmd/root.go` aligns with the ports defined in the docker compose for each network (docker compose is defined in the Stride repo):
  * Gaia: 26557
  * Juno: 26457
  * Osmo: 26357
* Point the firehose setup binaries in `scripts/setup.sh` to the dockernet paths
```
GAIAD="$STRIDE_REPO/build/gaiad --home $STRIDE_REPO/scripts/state/gaia1"
OSMOSISD="$STRIDE_REPO/build/osmosisd --home $STRIDE_REPO/scripts/state/osmo1"
JUNOD="$STRIDE_REPO/build/junod --home $STRIDE_REPO/scripts/state/juno1"
```

## Documentation and Release Note
  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no) no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no) no
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  / README.md  /   not documented) `scripts/README.md`
  - Does this pull request update existing proto field values (and require a backend and frontend migration)? (yes / no) no
  - Does this pull request change existing proto field names (and require a frontend migration)? (yes / no) no
